File: SDKs/googleapis.com/dfareporting/v3.3/python/src/sdk/models/shared/accountpermissiongroup.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclass
class AccountPermissionGroup:
    r"""AccountPermissionGroup
    AccountPermissionGroups contains a mapping of permission group IDs to names. A permission group is a grouping of account permissions.
    """
    
    id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    kind: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('kind') }})
    name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    

File: SDKs/googleapis.com/dfareporting/v4/typescript/src/sdk/models/shared/accountpermissiongroup.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// AccountPermissionGroup
/** 
 * AccountPermissionGroups contains a mapping of permission group IDs to names. A permission group is a grouping of account permissions.
**/
export class AccountPermissionGroup extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=kind" })
  kind?: string;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: string;
}
